### PR TITLE
Phase 1 of Edit Submission

### PR DESCRIPTION
**Pull request by dhaselhan**
* Rename Application -> ApplicationSubmission
* Create new front end and back end modules for ALCS Edit Feature
* Clean up existing edit module to export and create base step component to DRY things up
* Re-hook up API and Queries with new permission checks and uuid lookups
* Add URLs to go Portal <-> ALCS
* Change submissions and documents to be ManyToOne so we can clone them

### CODE REVIEW

File: alcs-frontend/src/environments/environment.prod.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export const environment = {
   apiUrl: (window.env?.['apiUrl'] || 'http://localhost:8080') + '/alcs',
   authUrl: window.env?.['apiUrl'] || 'http://localhost:8080',
   homeUrl: window.env?.['homeUrl'] || '/home',
+  portalUrl: window.env?.['portalUrl'] || 'http://localhost:4201',
```
Comment (mhuseinov) [inline alcs-frontend/src/environments/environment.prod.ts:13]: is it mapped on actual dev env?

File: portal-frontend/src/app/services/application-submission/application-submission-draft.service.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      this.overlayService.hideSpinner();
+    }
+
+    return undefined;
+  }
+
+  async publish(fileId: string) {
+    try {
+      this.overlayService.showSpinner();
+      // await firstValueFrom(
```
Comment (mhuseinov) [inline portal-frontend/src/app/services/application-submission/application-submission-draft.service.ts:58]: is this still relevant?

File: services/apps/alcs/src/portal/application-submission/application-submission.controller.spec.ts [hunk 3]
```diff
@@ -207,25 +214,29 @@ describe('ApplicationSubmissionController', () => {
...
           }),
         },
       },
-      'file-id',
+      '',
     );
 
     expect(application).toBeDefined();
-    expect(mockAppService.getForGovernmentByFileId).toHaveBeenCalledTimes(1);
+    //expect(mockLgService.getByGuid).toHaveBeenCalledTimes(1);
```
Comment (mhuseinov) [inline services/apps/alcs/src/portal/application-submission/application-submission.controller.spec.ts:238]: lets remove this

**Review by mhuseinov — APPROVED** [review]
👍

File: portal-frontend/src/app/services/application-submission/application-submission-draft.service.ts (hunk 2 above)
Comment (dhaselhan) [reply portal-frontend/src/app/services/application-submission/application-submission-draft.service.ts:58]: This is more placeholder code for the next ticket to implement publish

File: alcs-frontend/src/environments/environment.prod.ts (hunk 1 above)
Comment (dhaselhan) [reply alcs-frontend/src/environments/environment.prod.ts:13]: I have updated the template js files with the value and will make an MR on devops repo with the values

File: services/apps/alcs/src/portal/application-submission/application-submission.controller.spec.ts (hunk 3 above)
Comment (dhaselhan) [reply services/apps/alcs/src/portal/application-submission/application-submission.controller.spec.ts]: done

**Pull request merged**